funnel/models/event.py

```diff
@@ -0,0 +1,134 @@
+        for only the workshop or a single day of the conference.
+        An event is associated with multiple ticket types,
+        which helps make the distinction between participants
+    """
+    __tablename__ = 'event'
+
+    name = db.Column(db.Unicode(80), nullable=False)
+    proposal_space_id = db.Column(None, db.ForeignKey('proposal_space.id'), nullable=False)
+    proposal_space = db.relationship(ProposalSpace,
+        backref=db.backref('events', cascade='all, delete-orphan', lazy='dynamic'))
+    ticket_types = db.relationship("TicketType", secondary=EVENT_TICKET_TYPE_TABLE_NAME, lazy='dynamic')
+    participants = db.relationship("Participant", secondary=ATTENDEE_TABLE_NAME, lazy='dynamic')
+
+
+class TicketType(BaseMixin, db.Model):
+    """ Models different types of tickets. Eg: Early Geek, Super Early Geek, Workshop A.
+        A ticket type is associated with multiple events.
+    """
+    __tablename__ = 'ticket_type'
+
+    name = db.Column(db.Unicode(80), nullable=False)
+    proposal_space_id = db.Column(None, db.ForeignKey('proposal_space.id'), nullable=False)
+    proposal_space = db.relationship(ProposalSpace,
+        backref=db.backref('ticket_types', cascade='all, delete-orphan', lazy='dynamic'))
+    events = db.relationship("Event", secondary=EVENT_TICKET_TYPE_TABLE_NAME)
+
+
+class Participant(BaseMixin, db.Model):
+    """ Model users participating in the proposal space
+        as an attendee, speaker, volunteer, sponsor.
+    """
+    __tablename__ = 'participant'
+
+    fullname = db.Column(db.Unicode(80), nullable=False)
+    #: Unvalidated email address
+    email = db.Column(db.Unicode(80), nullable=False)
+    #: Unvalidated phone number
+    phone = db.Column(db.Unicode(80), nullable=True)
+    #: Unvalidated Twitter id
+    twitter = db.Column(db.Unicode(80), nullable=True)
+    #: Job title
+    job_title = db.Column(db.Unicode(80), nullable=True)
+    #: Company
+    company = db.Column(db.Unicode(80), nullable=True)
+    #: Participant's city
+    city = db.Column(db.Unicode(80), nullable=True)
+    # public key
+    puk = db.Column(db.Unicode(44), nullable=False, default=make_public_key, unique=True)
+    #: Access key for connecting to the user record
+    key = db.Column(db.Unicode(44), nullable=False, default=make_private_key, unique=True)
+    badge_printed = db.Column(db.Boolean, default=False, nullable=False)
+    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=True)
+    user = db.relationship(User, primaryjoin=user_id == User.id,
+        backref=db.backref('participant', cascade="all, delete-orphan"))
+    proposal_space_id = db.Column(db.Integer, db.ForeignKey('proposal_space.id'), nullable=False)
+    proposal_space = db.relationship(ProposalSpace,
+        backref=db.backref('participants', cascade='all, delete-orphan', lazy='dynamic'))
+    events = association_proxy('attendees', 'event')
+
+    __table_args__ = (db.UniqueConstraint("email", "proposal_space_id"), {})
```

Comment:
The `{}` is unnecessary with current versions of SQLAlchemy (it was required in the past)